xla/stream_executor/gpu/gpu_executor.h

```diff
@@ -186,7 +186,7 @@ class GpuExecutor : public StreamExecutorCommon {
 
   absl::StatusOr<MemorySpace> GetPointerMemorySpace(const void* ptr) override {
     return GpuDriver::GetPointerMemorySpace(
-        reinterpret_cast<GpuDevicePtr>(ptr));
+        const_cast<GpuDevicePtr>(ptr));
```

Comment:
With this change, we get a compile error internally:

error: const_cast to 'GpuDevicePtr' (aka 'unsigned long long'), which is not a reference, pointer-to-object, or pointer-to-data-member

So it seems that GpuDevicePtr is not a good abstraction over the differences between CUDA and ROCM

Reply:
One thing we could try is:

reinterpret_cast<GpuDevicePtr>(const_cast<void*>(ptr));

I see this in gpu_stream.cc

Reply:
Fixed, hope it's okay now